### PR TITLE
media: add toString ability to createBreakpoints

### PR DESCRIPTION
**Pull request by atk**

### CODE REVIEW

File: packages/media/README.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Collection of reactive primitives to deal with media queries.
 - [`createMediaQuery`](#createMediaQuery) - Creates a very simple and straightforward media query monitor.
 - [`createBreakpoints`](#createBreakpoints) - Creates a multi-breakpoint monitor to make responsive components easily.
 - [`createPrefersDark`](#createPrefersDark) - Provides a signal indicating if the user has requested dark color theme.
+- [`sortBreakpoints`](#sortBreakpoints) - Sorts the breakpoints from small to large (to enable correct string coercion of the output of `createBreakpoints`).
```
Comment (thetarnav) [inline packages/media/README.md:18]: it's a part of `createMediaQuery` API, not a standalone
```suggestion
```

**Review by thetarnav — CHANGES_REQUESTED** [review]
It seems fair to add this
But I don't think it should be done with `.toString()`, but with another, custom method that we can control the type of. Or better, a separate function like `getMathingBreakpoint()`.

image

File: packages/media/src/index.ts [hunk 2]
```diff
@@ -99,11 +99,14 @@ export interface BreakpointOptions<T extends Breakpoints> {
...
-  return matches;
-};
+const getEmptyMatchesFromBreakpoints = <T extends Breakpoints>(breakpoints: T): Matches<T> =>
+  entries(breakpoints).reduce<Record<keyof T, boolean>>(
+    (matches, [key]: [key: keyof T, value: string]) => {
+      matches[key] = false;
+      return matches;
+    },
+    {} as Record<keyof T, boolean>,
+  ) as Matches<T>;
```
[the remark this answers is not in the document]
Comment (atk) [reply packages/media/src/index.ts:109]: Sorry. Using `forEach` instead of `reduce` to reduce an array is a pet peeve of mine.

**Comment by atk** [comment]
> It seems fair to add this But I don't think it should be done with `.toString()`, but with another, custom method that we can control the type of. Or better, a separate function like `getMathingBreakpoint()`.

Why would we need to control the type of? What we want is a string, which is exactly what toString is supposed to deliver: some more or less useful string representation of the contents. Arrays set a precedent: try `[1,2,3]+''` – what you'll get is exactly what .toString() is meant for.

The most useful short string representation of our `matches` object is the name of the largest matching breakpoint. Thus, this is a prime example of .toString done right.

**Comment by thetarnav** [comment]
We need to control the type, to be able to return `keyof T` instead of `string` to be able to use it as a index

**Comment by atk** [comment]
Now I get your point. I'll consider it.

**Comment by atk** [comment]
I thought long and hard about that and at the end, I settled on a "key" property, because that is the shortest name I could come up with that still conveys the intended meaning.

**Comment by atk** [comment]
fixed it.

**Pull request merged**